### PR TITLE
kile-wl: 2021-09-30 → 2023-07-23

### LINKED ISSUE #248408: Update request: kile-wl 2021-09-30 → 2023-07-23 (c2420876)
- Package name: `kile-wl`
- Latest released version: git commit `c24208761d04e0a74d203fc1dcd2f7fed68da388` (2023-07-23)

- Current version on the unstable channel: git commit `b543d435b92498b72609a05048bc368837a7b455` (2021-09-29)
- Current version on the stable/release channel: git commit `b543d435b92498b72609a05048bc368837a7b455` (2021-09-29)

- [X] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@fortuneteller2k

**Additional notes**
- The package is severely outdated, and the kile configuration format has changed (effectively, it's ["kile 2.0"](https://gitlab.com/snakedye/kile/-/commit/625f91010b920587dbf0ee23113eb8aa51cc6ec3))
- The project's `Cargo.toml` has a git dependency.
 
-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by dithpri**
## Description of changes
Fixes #248408

Upstream changes:
- **Breaking change**: different configuration format
- No docs/manpages provided, instead,  there's a [project wiki](https://gitlab.com/snakedye/kile/-/wikis/Home)
- [Full commit list & diff](https://gitlab.com/snakedye/kile/-/compare/b543d435b92498b72609a05048bc368837a7b455...c24208761d04e0a74d203fc1dcd2f7fed68da388?from_project_id=23400985&straight=false)

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/kile-wl/default.nix [hunk 1]
```diff
@@ -1,28 +1,26 @@
-{ lib, fetchFromGitLab, unstableGitUpdater, rustPlatform, scdoc }:
+{ lib, fetchFromGitLab, unstableGitUpdater, rustPlatform }:
```
Comment (cafkafk) [inline pkgs/applications/misc/kile-wl/default.nix:2]:
```suggestion
{
  lib,
  fetchFromGitLab,
  unstableGitUpdater,
  rustPlatform,
}:
```

File: pkgs/applications/misc/kile-wl/default.nix [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
-  version = "2.0";
+  version = "unstable-2023-07-23";
 
   src = fetchFromGitLab {
     owner = "snakedye";
     repo = "kile";
-    rev = "b543d435b92498b72609a05048bc368837a7b455";
-    sha256 = "sha256-+SjdhSRT6TGbwvgZti8t9wYJx8LEtY3pleDZx/AEkio=";
+    rev = "c24208761d04e0a74d203fc1dcd2f7fed68da388";
+    sha256 = "sha256-4iclNVd7nm6LkgvsHwWaWyi1bZL/A+bbT5OSXn70bLs=";
```
Comment (cafkafk) [inline pkgs/applications/misc/kile-wl/default.nix:11]:
```suggestion
    hash = "sha256-4iclNVd7nm6LkgvsHwWaWyi1bZL/A+bbT5OSXn70bLs=";
```

**Review by cafkafk — COMMENTED** [review]
Here are some stylistic suggestions, but they shouldn't block merging this.

**Review by moni-dz — APPROVED** [review]

**Comment by tomberek** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/250957?

**Comment by scotabroad** [comment]
> Superseded by #250957?

No. That pull request failed to fix [#248408](https://github.com/NixOS/nixpkgs/issues/248408) as it only changed the cargo hash, but not the gitlab hash, meaning it was still pulling the 2021 code. This pull request will update the gitlab hash to the latest version.

**Pull request merged**